File: backend/app/schemas/vendorPayment.py
from pydantic import BaseModel
from typing import Optional, List, Literal
from datetime import datetime

PaymentStatusBayar = Literal["belum_bayar", "lunas"]


class VendorPaymentSchema(BaseModel):
    id: int
    session_id: str
    vendor_id: int
    vendor_nama: str
    judul: str
    nominal: int
    tanggal_jatuh_tempo: Optional[datetime]
    status_bayar: PaymentStatusBayar
    tanggal_dibayar: Optional[datetime]
    catatan: Optional[str]

    class Config:
        from_attributes = True


class VendorPaymentCreateRequest(BaseModel):
    session_id: str
    vendor_id: int
    judul: str
    nominal: int
    tanggal_jatuh_tempo: Optional[str] = None
    catatan: Optional[str] = None


class VendorPaymentUpdateRequest(BaseModel):
    judul: Optional[str] = None
    nominal: Optional[int] = None
    tanggal_jatuh_tempo: Optional[str] = None
    status_bayar: Optional[PaymentStatusBayar] = None
    tanggal_dibayar: Optional[str] = None
    catatan: Optional[str] = None


class VendorPaymentSummary(BaseModel):
    total_terjadwal: int 
    total_lunas: int
    total_belum_lunas: int
    jumlah_termin_lunas: int
    jumlah_termin_belum_lunas: int

File: frontend/src/types/vendorPayment.ts
export type PaymentStatusBayar = "belum_bayar" | "lunas";

export interface VendorPayment {
  id: number;
  session_id: string;
  vendor_id: number;
  vendor_nama: string;
  judul: string;
  nominal: number;
  tanggal_jatuh_tempo: string | null;
  status_bayar: PaymentStatusBayar;
  tanggal_dibayar: string | null;
  catatan: string | null;
}

export interface VendorPaymentCreateRequest {
  session_id: string;
  vendor_id: number;
  judul: string;
  nominal: number;
  tanggal_jatuh_tempo?: string | null;
  catatan?: string | null;
}

export interface VendorPaymentUpdateRequest {
  judul?: string;
  nominal?: number;
  tanggal_jatuh_tempo?: string | null;
  status_bayar?: PaymentStatusBayar;
  tanggal_dibayar?: string | null;
  catatan?: string | null;
}

export interface VendorPaymentSummary {
  total_terjadwal: number;
  total_lunas: number;
  total_belum_lunas: number;
  jumlah_termin_lunas: number;
  jumlah_termin_belum_lunas: number;
}
